xarray/core/dataset.py

```diff
@@ -4127,18 +4119,6 @@ def interp(
 
         coords = either_dict_or_kwargs(coords, coords_kwargs, "interp")
         indexers = dict(self._validate_interp_indexers(coords))
-
-        if coords:
```

Comment:
Handled by `vectorize=True`. This is possibly a perf regression with numpy arrays, but a massive improvement with chunked arrays.

Reply:
For posterity the bad thing about this approach is that it can greatly expand the number of core dimensions for the problem, limiting the potential for parallelism.

Consider the problem in https://github.com/pydata/xarray/issues/6799#issuecomment-2474126217. In the following, dimension names are listed out in `[]`. 

`da[time, q, lat, lon].interp(q=bar[lat,lon])` gets rewritten to `da[time,q,lat,lon].interp(q=bar[lat, lon], lat=lat[lat], lon=lon[lon])` which thanks to our automatic rechunking, makes dask merge chunks in `lat, lon` too, for no benefit.